sys/crypto/modes/cbc.c

```diff
@@ -52,18 +60,63 @@ int cipher_encrypt_cbc(const cipher_t *cipher, uint8_t iv[16],
     return offset;
 }
 
+int cipher_encrypt_cbc_with_padding(const cipher_t *cipher, uint8_t iv[16],
+                                    const uint8_t *input, size_t length, uint8_t *output, uint8_t padding_type)
+{
+    /* Only PKCS7 is supported */
+    if (padding_type != PADDING_TYPE_PKCS7) {
```

Comment:
I'm not sure, actually. PKCS#7 seems to be the one that is used the most, but there are a few others, including ISO standards.
This would keep this API open for possible future developments.

Reply:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) :wink: 

And don't you think `cipher_encrypt_cbc_pkcs7()` would feel like a better API? It could still be turned into a `static inline` wrapper function if that need arises in the future.

Reply:
Sure, what would a generic `pkcs7_padding()` function even be used for?

Reply:
done